### ISSUE TITLE
Is it possible to use a stateful LSTM with a long time series

### ISSUE THREAD

**Issue by peschn** [keras-team/keras#7996]
I have a data set consisting of variable length audio sequences (30 secs to 1.5h). Each of the sequences has a label assigned, that I want to predict with a neural network with multiple LSTM layers. The challenge is, that I can't feed the long sequences at once into the model, as memory doesn't allow this. My thoughts are the following on how to solve this problem:

1. Break up the long sequences into smaller sequences (1 sec) and assign the label the is associated with the long sequence. Example: 

long Sequence : [1, 2, 3, 4, 5, 6] --> Label
after breaking it down: [1,2,3] --> Label, [4,5,6] --> Label

2. Feed the sequences into the model, but make the LSTMs stateful. States are reset once an entire file is processed (e.g. after [1,2.3] and [4,5,6] are processed).

My questions is now, whether this is an acceptable procedure? My concern is, that the model will not learn the long term dependencies.

**Comment by td2014** [comment]
@peschn .  Setting aside the memory limitation for a moment, from a theoretical perspective would your multilayer LSTM model be able to learn the structure of the input sequence-or said another way, how much does long term dependency really matter in the desired classification rate?  Perhaps the level of expected fidelity with full information (ie. how accurate the model really needs to be) might help guide which choices will be best for the memory limited approach.  I hope this helps suggest some direction.  Thanks.

**Comment by peschn** [comment]
@td2014 thanks a lot for your answer. 

> how much does long term dependency really matter in the desired classification rate

I am not sure. The audio file contains a conversation and I suspect that only parts of it are important to predict the label. But it is not clear which part of the conversation it is, which makes it a bit tricky.

Do you think, a stateful LSTM will be capable of learning longer dependencies, i.e. that go beyond the timesteps in a batch?

**Comment by viswaravi** [comment]
@peschn I'm also working on a similar problem and thought of the same way to train the model. Have you succeeded in your work?
